clients/web/src/rest.js

```diff
@@ -5,28 +5,75 @@ import Backbone from 'backbone';
 import events from 'girder/events';
 import { getCurrentToken, cookie } from 'girder/auth';
 
-var apiRoot = $('#g-global-info-apiroot').text().replace('%HOST%', window.location.origin) || '/api/v1';
-var staticRoot = $('#g-global-info-staticroot').text().replace('%HOST%', window.location.origin) || '/static';
+let apiRoot;
+let staticRoot;
 var uploadHandlers = {};
 var uploadChunkSize = 1024 * 1024 * 64; // 64MB
 
 /**
+ * Get the root path to the API.
+ *
+ * This may be an absolute path, or a path relative to the application root. It will never include
+ * a trailing slash.
+ *
+ * @returns {string}
+ */
+function getApiRoot() {
+    return apiRoot;
+}
+
+/**
  * Set the root path to the API.
  *
- * @param root The full root path for the API.
+ * @param {string} root The root path for the API.
  */
 function setApiRoot(root) {
-    apiRoot = root;
+    // Strip trailing slash
+    apiRoot = root.replace(/\/$/, '');
+}
+
+/**
+ * Get the root path to the static content.
+ *
+ * This may be an absolute path, or a path relative to the application root. It will never include
+ * a trailing slash.
+ *
+ * @returns {string}
+ */
+function getStaticRoot() {
+    return staticRoot;
 }
 
-/** Set the root path to the static content.
+/**
+ * Set the root path to the static content.
  *
- * @param root The full root path for the static content.
+ * @param {string} root The root path for the static content.
  */
 function setStaticRoot(root) {
-    staticRoot = root;
+    // Strip trailing slash
+    staticRoot = root.replace(/\/$/, '');
+    // publicPath would normally be set in the Webpack config file, but is not known at compile
+    // time.
+    __webpack_public_path__ = `${getStaticRoot()}/built/`; // eslint-disable-line no-undef, camelcase
```

Comment:
This is the correct way to do things. The CSS is being loaded by the Browser's native mechanism for handling `url()` in CSS, [which allows relative paths in the spec](https://developer.mozilla.org/en-US/docs/Web/CSS/url).

In Javascript (whether its in native JS or a Pug file), there is no native concept of a relative path: JS files are run the exact same way, regardless of how they were sourced or loaded by the browser. So, if you want to load static assets via Webpack in JS, you need to have `publicPath` set, because the path that the files were emitted to on disk is not necessarily connected to the URL that the files can be fetched from.

It's definitely a bug that `publicPath` was never set before. The extensive caviet comment is a result of the fact that ES6 imports are always hoisted to the top of a module (unlike `require`), in combination with the fact that Girder's architecture decided to allow `staticRoot` to be set at any point at runtime.